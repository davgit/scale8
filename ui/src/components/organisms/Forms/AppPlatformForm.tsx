import { FC } from 'react';
import DrawerFormLayout from '../../molecules/DrawerFormLayout';
import {
    AppPlatformFormProps,
    AppPlatformValues,
} from '../../../dialogPages/tagManager/app/InstallAppPlatform';
import { DialogFormContextProvider } from '../../../context/DialogFormContext';
import { DialogFormSelect } from '../../atoms/DialogFormInputs/DialogFormSelect';

const AppPlatformForm: FC<AppPlatformFormProps> = (props: AppPlatformFormProps) => {
    const notAvailable = props.availableAppPlatforms.length < 1;

    return (
        <DialogFormContextProvider<AppPlatformValues> formProps={props}>
            <DrawerFormLayout {...props}>
                {notAvailable ? (
                    <small>There are no more platforms available.</small>
                ) : (
                    <DialogFormSelect
                        label="Platform"
                        name="platformId"
                        values={props.availableAppPlatforms}
                    />
                )}
            </DrawerFormLayout>
        </DialogFormContextProvider>
    );
};

export default AppPlatformForm;
