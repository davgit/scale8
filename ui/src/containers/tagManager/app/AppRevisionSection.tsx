import { FC } from 'react';
import { useQuery } from '@apollo/client';
import TagIcon from '../../../components/atoms/Icons/TagIcon';
import PlatformRevisionIcon from '../../../components/atoms/Icons/PlatformRevisionIcon';
import NavAppRevisionQuery from '../../../gql/queries/NavAppRevisionQuery';
import { NavAppRevision } from '../../../gql/generated/NavAppRevision';
import {
    buildAppButtonProps,
    buildAppRevisionButtonProps,
    buildOrgButtonProps,
    buildTagManagerButtonProps,
} from '../../../utils/BreadcrumbButtonsUtils';
import { Section, SectionProps } from '../../abstractions/Section';
import ActionIcon from '../../../components/atoms/Icons/ActionIcon';
import TriggerIcon from '../../../components/atoms/Icons/TriggerIcon';
import { buildAppRevisionBreadcrumbActions } from '../../../utils/BuildAppRevisionBreadcrumbActions';
import { extractPermissionsFromOrgUser } from '../../../context/OrgUserReducer';
import { SectionKey } from '../../SectionsDetails';
import { useLoggedInState } from '../../../context/AppContext';
import { useRouter } from 'next/router';
import { ChildrenAndIdProps } from '../../../types/props/ChildrenAndIdProps';
import { toAppRevision } from '../../../utils/NavigationPaths';
import { analyticsEnabled } from '../../../utils/AnalyticsUtils';

const AppRevisionSection: FC<ChildrenAndIdProps> = (props: ChildrenAndIdProps) => {
    const router = useRouter();

    const { id, children } = props;

    const { orgUserState, templateInteractions } = useLoggedInState();
    const { ask, dispatchDialogAction, setRefreshCurrentPage } = templateInteractions;

    const currentOrgPermissions = extractPermissionsFromOrgUser(orgUserState);

    const sectionProps: SectionProps<NavAppRevision> = {
        children,
        sectionKey: SectionKey.appRevision,
        queryResult: useQuery<NavAppRevision>(NavAppRevisionQuery, {
            variables: { id },
        }),
        initContext: (data) => {
            templateInteractions.setSectionHasAnalytics(analyticsEnabled(data.getRevision.app));
        },
        buildButtonsProps: (data) => [
            buildOrgButtonProps(
                router,
                data.me.orgs,
                data.getRevision.app.tag_manager_account.org.id,
                data.getRevision.app.tag_manager_account.org.name,
            ),
            buildTagManagerButtonProps(
                router,
                data.getRevision.app.tag_manager_account.id,
                data.getRevision.app.tag_manager_account.org.data_manager_account?.id ?? '',
            ),
            buildAppButtonProps(
                router,
                data.getRevision.app.tag_manager_account.apps,
                data.getRevision.app.id,
                data.getRevision.app.name,
            ),
            buildAppRevisionButtonProps(
                router,
                data.getRevision.app.revisions,
                id,
                data.getRevision.name,
                true,
            ),
        ],
        buildMenuItemsProps: () => [
            {
                icon: () => <TagIcon />,
                label: 'Tags',
                link: toAppRevision({ id }, 'tags'),
            },
            {
                icon: () => <TriggerIcon />,
                label: 'Global Triggers',
                link: toAppRevision({ id }, 'global-triggers'),
            },
            {
                icon: () => <ActionIcon />,
                label: 'Global Actions',
                link: toAppRevision({ id }, 'global-actions'),
            },
            {
                icon: () => <PlatformRevisionIcon />,
                label: 'Platform Revisions',
                link: toAppRevision({ id }, 'app-platform-revisions'),
            },
        ],
        extractOrgUserDetails: (data) => data.getRevision.app.tag_manager_account.org,
        buildBreadcrumbActions: (data) =>
            buildAppRevisionBreadcrumbActions(
                id,
                data.getRevision.name,
                router,
                currentOrgPermissions,
                dispatchDialogAction,
                setRefreshCurrentPage,
                ask,
                data.getRevision.locked,
            ),
        accountExpireIn: orgUserState?.tagManagerAccount?.trialExpiration ?? undefined,
        accountIsTrial: orgUserState?.tagManagerAccount?.isTrial ?? undefined,
    };

    return <Section<NavAppRevision> {...sectionProps} />;
};

export default AppRevisionSection;
