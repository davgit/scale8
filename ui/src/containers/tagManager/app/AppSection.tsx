import { FC } from 'react';
import { useQuery } from '@apollo/client';
import { NavApp } from '../../../gql/generated/NavApp';
import NavAppQuery from '../../../gql/queries/NavAppQuery';
import RevisionIcon from '../../../components/atoms/Icons/RevisionIcon';
import EnvironmentIcon from '../../../components/atoms/Icons/EnvironmentIcon';
import PlatformIcon from '../../../components/atoms/Icons/PlatformIcon';
import {
    buildAppButtonProps,
    buildOrgButtonProps,
    buildTagManagerButtonProps,
} from '../../../utils/BreadcrumbButtonsUtils';
import { Section, SectionProps } from '../../abstractions/Section';
import AppAnalyticsIcon from '../../../components/atoms/Icons/AppAnalyticsIcon';
import { SectionKey } from '../../SectionsDetails';
import { useLoggedInState } from '../../../context/AppContext';
import { useRouter } from 'next/router';
import { ChildrenAndIdProps } from '../../../types/props/ChildrenAndIdProps';
import { toApp } from '../../../utils/NavigationPaths';
import { analyticsEnabled } from '../../../utils/AnalyticsUtils';
import { SideMenuButtonProps } from '../../../components/molecules/SideMenuButton';

const AppSection: FC<ChildrenAndIdProps> = (props: ChildrenAndIdProps) => {
    const router = useRouter();

    const { id, children } = props;

    const { orgUserState, templateInteractions } = useLoggedInState();

    const sectionProps: SectionProps<NavApp> = {
        children,
        sectionKey: SectionKey.app,
        queryResult: useQuery<NavApp>(NavAppQuery, {
            variables: { id },
        }),
        initContext: (data) => {
            templateInteractions.setSectionHasAnalytics(analyticsEnabled(data.getApp));
        },
        buildButtonsProps: (data) => [
            buildOrgButtonProps(
                router,
                data.me.orgs,
                data.getApp.tag_manager_account.org.id,
                data.getApp.tag_manager_account.org.name,
            ),
            buildTagManagerButtonProps(
                router,
                data.getApp.tag_manager_account.id,
                data.getApp.tag_manager_account.org.data_manager_account?.id ?? '',
            ),
            buildAppButtonProps(
                router,
                data.getApp.tag_manager_account.apps,
                id,
                data.getApp.name,
                true,
            ),
        ],
        buildMenuItemsProps: (_, data) => [
            ...((analyticsEnabled(data.getApp)
                ? [
                      {
                          icon: <AppAnalyticsIcon />,
                          label: 'Analytics',
                          link: toApp({ id }, 'analytics'),
                      },
                  ]
                : []) as SideMenuButtonProps[]),
            {
                icon: <RevisionIcon />,
                label: 'Revisions',
                link: toApp({ id }, 'revisions'),
            },
            {
                icon: <EnvironmentIcon />,
                label: 'Environments',
                link: toApp({ id }, 'environments'),
            },
            {
                icon: <PlatformIcon />,
                label: 'Installed Platforms',
                link: toApp({ id }, 'platforms'),
            },
        ],
        extractOrgUserDetails: (data) => data.getApp.tag_manager_account.org,
        accountExpireIn: orgUserState?.tagManagerAccount?.trialExpiration ?? undefined,
        accountIsTrial: orgUserState?.tagManagerAccount?.isTrial ?? undefined,
    };

    return <Section<NavApp> {...sectionProps} />;
};

export default AppSection;
