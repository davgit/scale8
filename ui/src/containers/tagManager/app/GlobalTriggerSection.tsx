import { FC } from 'react';
import { useQuery } from '@apollo/client';
import NavGlobalTriggerQuery from '../../../gql/queries/NavGlobalTriggerQuery';
import { NavGlobalTrigger } from '../../../gql/generated/NavGlobalTrigger';
import {
    buildAppButtonProps,
    buildAppRevisionButtonProps,
    buildGlobalTriggerButtonProps,
    buildOrgButtonProps,
    buildTagManagerButtonProps,
} from '../../../utils/BreadcrumbButtonsUtils';
import { Section, SectionProps } from '../../abstractions/Section';
import { buildAppRevisionBreadcrumbActions } from '../../../utils/BuildAppRevisionBreadcrumbActions';
import { extractPermissionsFromOrgUser } from '../../../context/OrgUserReducer';
import { SectionKey } from '../../SectionsDetails';
import { useLoggedInState } from '../../../context/AppContext';
import { useRouter } from 'next/router';
import { ChildrenAndIdProps } from '../../../types/props/ChildrenAndIdProps';
import { analyticsEnabled } from '../../../utils/AnalyticsUtils';

const GlobalTriggerSection: FC<ChildrenAndIdProps> = (props: ChildrenAndIdProps) => {
    const router = useRouter();

    const { id, children } = props;

    const { orgUserState, templateInteractions } = useLoggedInState();

    const { ask, dispatchDialogAction, setRefreshCurrentPage } =
        templateInteractions;
    const currentOrgPermissions = extractPermissionsFromOrgUser(orgUserState);

    const sectionProps: SectionProps<NavGlobalTrigger> = {
        children,
        sectionKey: SectionKey.globalTrigger,
        queryResult: useQuery<NavGlobalTrigger>(NavGlobalTriggerQuery, {
            variables: { id },
        }),
        initContext: (data) => {
            templateInteractions.setSectionHasAnalytics(
                analyticsEnabled(data.getTrigger.revision.app),
            );
        },
        buildButtonsProps: (data) => [
            buildOrgButtonProps(
                router,
                data.me.orgs,
                data.getTrigger.revision.app.tag_manager_account.org.id,
                data.getTrigger.revision.app.tag_manager_account.org.name,
            ),
            buildTagManagerButtonProps(
                router,
                data.getTrigger.revision.app.tag_manager_account.id,
                data.getTrigger.revision.app.tag_manager_account.org.data_manager_account?.id ?? '',
            ),
            buildAppButtonProps(
                router,
                data.getTrigger.revision.app.tag_manager_account.apps,
                data.getTrigger.revision.app.id,
                data.getTrigger.revision.app.name,
            ),
            buildAppRevisionButtonProps(
                router,
                data.getTrigger.revision.app.revisions,
                data.getTrigger.revision.id,
                data.getTrigger.revision.name,
            ),
            buildGlobalTriggerButtonProps(
                router,
                data.getTrigger.revision.global_triggers,
                id,
                data.getTrigger.name,
                true,
            ),
        ],
        buildMenuItemsProps: () => [],
        extractOrgUserDetails: (data) => data.getTrigger.revision.app.tag_manager_account.org,
        buildBreadcrumbActions: (data) =>
            buildAppRevisionBreadcrumbActions(
                data.getTrigger.revision.id,
                data.getTrigger.revision.name,
                router,
                currentOrgPermissions,
                dispatchDialogAction,
                setRefreshCurrentPage,
                ask,
                data.getTrigger.revision.locked,
            ),
        accountExpireIn: orgUserState?.tagManagerAccount?.trialExpiration ?? undefined,
        accountIsTrial: orgUserState?.tagManagerAccount?.isTrial ?? undefined,
    };

    return <Section<NavGlobalTrigger> {...sectionProps} />;
};

export default GlobalTriggerSection;
