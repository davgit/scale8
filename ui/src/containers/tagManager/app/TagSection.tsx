import { FC } from 'react';
import { useQuery } from '@apollo/client';
import NavTagQuery from '../../../gql/queries/NavTagQuery';
import { NavTag } from '../../../gql/generated/NavTag';
import {
    buildAppButtonProps,
    buildAppRevisionButtonProps,
    buildOrgButtonProps,
    buildTagButtonProps,
    buildTagManagerButtonProps,
} from '../../../utils/BreadcrumbButtonsUtils';
import { Section, SectionProps } from '../../abstractions/Section';
import { buildAppRevisionBreadcrumbActions } from '../../../utils/BuildAppRevisionBreadcrumbActions';
import { extractPermissionsFromOrgUser } from '../../../context/OrgUserReducer';
import { SectionKey } from '../../SectionsDetails';
import { useLoggedInState } from '../../../context/AppContext';
import { useRouter } from 'next/router';
import { ChildrenAndIdProps } from '../../../types/props/ChildrenAndIdProps';
import { analyticsEnabled } from '../../../utils/AnalyticsUtils';

const TagSection: FC<ChildrenAndIdProps> = (props: ChildrenAndIdProps) => {
    const router = useRouter();

    const { id, children } = props;

    const { orgUserState, templateInteractions } = useLoggedInState();

    const { ask, dispatchDialogAction, setRefreshCurrentPage } =
        templateInteractions;
    const currentOrgPermissions = extractPermissionsFromOrgUser(orgUserState);

    const sectionProps: SectionProps<NavTag> = {
        children,
        sectionKey: SectionKey.tag,
        queryResult: useQuery<NavTag>(NavTagQuery, {
            variables: { id },
        }),
        initContext: (data) => {
            templateInteractions.setSectionHasAnalytics(analyticsEnabled(data.getTag.revision.app));
        },
        buildButtonsProps: (data) => [
            buildOrgButtonProps(
                router,
                data.me.orgs,
                data.getTag.revision.app.tag_manager_account.org.id,
                data.getTag.revision.app.tag_manager_account.org.name,
            ),
            buildTagManagerButtonProps(
                router,
                data.getTag.revision.app.tag_manager_account.id,
                data.getTag.revision.app.tag_manager_account.org.data_manager_account?.id ?? '',
            ),
            buildAppButtonProps(
                router,
                data.getTag.revision.app.tag_manager_account.apps,
                data.getTag.revision.app.id,
                data.getTag.revision.app.name,
            ),
            buildAppRevisionButtonProps(
                router,
                data.getTag.revision.app.revisions,
                data.getTag.revision.id,
                data.getTag.revision.name,
            ),
            buildTagButtonProps(router, data.getTag.revision.tags, id, data.getTag.name, true),
        ],
        buildMenuItemsProps: () => [],
        extractOrgUserDetails: (data) => data.getTag.revision.app.tag_manager_account.org,
        buildBreadcrumbActions: (data) =>
            buildAppRevisionBreadcrumbActions(
                data.getTag.revision.id,
                data.getTag.revision.name,
                router,
                currentOrgPermissions,
                dispatchDialogAction,
                setRefreshCurrentPage,
                ask,
                data.getTag.revision.locked,
            ),
        accountExpireIn: orgUserState?.tagManagerAccount?.trialExpiration ?? undefined,
        accountIsTrial: orgUserState?.tagManagerAccount?.isTrial ?? undefined,
    };

    return <Section<NavTag> {...sectionProps} />;
};

export default TagSection;
