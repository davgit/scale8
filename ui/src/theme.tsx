import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
