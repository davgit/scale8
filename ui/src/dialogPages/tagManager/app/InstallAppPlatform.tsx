import { FC } from 'react';
import { FormProps, FormValidationResult } from '../../../hooks/form/useFormValidation';
import { ApolloError, useMutation, useQuery } from '@apollo/client';
import AppPlatformForm from '../../../components/organisms/Forms/AppPlatformForm';
import FetchAvailableAppPlatformsQuery from '../../../gql/queries/FetchAvailableAppPlatformsQuery';
import { FetchAvailableAppPlatforms } from '../../../gql/generated/FetchAvailableAppPlatforms';
import InstallAppPlatformQuery from '../../../gql/mutations/InstallAppPlatformQuery';
import { InstallPlatform } from '../../../gql/generated/InstallPlatform';
import { DialogPageProps } from '../../../types/DialogTypes';
import { DialogPreloadForm, DialogPreloadFormProps } from '../../abstractions/DialogPreloadForm';
import { buildStandardFormInfo } from '../../../utils/InfoLabelsUtils';

export type AppPlatformValues = {
    platformId: string;
};

export type AppPlatformFormProps = FormProps<AppPlatformValues> & {
    availableAppPlatforms: { key: string; text: string }[];
};

const InstallAppPlatform: FC<DialogPageProps> = (props: DialogPageProps) => {
    const appId = props.contextId;

    const installAppPlatformProps: DialogPreloadFormProps<
        FetchAvailableAppPlatforms,
        AppPlatformValues,
        AppPlatformFormProps,
        InstallPlatform
    > = {
        loadQuery: useQuery<FetchAvailableAppPlatforms>(FetchAvailableAppPlatformsQuery, {
            variables: { id: appId },
        }),
        buildInitialStatePreload: () => ({
            platformId: '',
        }),
        saveQuery: useMutation<InstallPlatform>(InstallAppPlatformQuery),
        mapSaveData: (appPlatformValues: AppPlatformValues) => ({
            appInstallPlatformInput: {
                app_id: appId,
                platform_id: appPlatformValues.platformId,
            },
        }),
        buildFormProps: (
            formLoadedData: FetchAvailableAppPlatforms,
            formValidationValues: FormValidationResult<AppPlatformValues>,
            gqlError?: ApolloError,
        ) => ({
            ...formValidationValues,
            gqlError,
            submitText: 'Install Platform',
            title: 'Install Platform',
            availableAppPlatforms: formLoadedData.getPublicPlatforms
                .filter(
                    (_) =>
                        !formLoadedData.getApp.app_platforms
                            .map((p) => p.platform.id)
                            .includes(_.id),
                )
                .map((_) => ({
                    key: _.id,
                    text: _.name,
                })),
            formInfoProps: buildStandardFormInfo('appPlatforms', 'Create'),
            handleDialogClose: props.handleDialogClose,
        }),
        checkSuccessfullySubmitted: (formMutationData) =>
            formMutationData?.installPlatform.id !== undefined,
        pageComponent: AppPlatformForm,
        validators: [],
        ...props,
    };

    return (
        <DialogPreloadForm<
            FetchAvailableAppPlatforms,
            AppPlatformValues,
            AppPlatformFormProps,
            InstallPlatform
        >
            {...installAppPlatformProps}
        />
    );
};

export default InstallAppPlatform;
