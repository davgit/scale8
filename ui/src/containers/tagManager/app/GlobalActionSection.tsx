import { FC } from 'react';
import { useQuery } from '@apollo/client';
import { buildGlobalActionButtonProps } from '../../../utils/BreadcrumbButtonsUtils';
import { Section, SectionProps } from '../../abstractions/Section';
import NavGlobalActionQuery from '../../../gql/queries/NavGlobalActionQuery';
import { NavGlobalAction } from '../../../gql/generated/NavGlobalAction';
import { buildAppRevisionBreadcrumbActions } from '../../../utils/BuildAppRevisionBreadcrumbActions';
import { extractPermissionsFromOrgUser } from '../../../context/OrgUserReducer';
import { SectionKey } from '../../SectionsDetails';
import { useConfigState, useLoggedInState } from '../../../context/AppContext';
import { useRouter } from 'next/router';
import { ChildrenAndIdProps } from '../../../types/props/ChildrenAndIdProps';
import { analyticsEnabled, errorTrackingEnabled } from '../../../utils/AnalyticsUtils';
import { buildAppRevisionButtons } from './AppRevisionSection';

const GlobalActionSection: FC<ChildrenAndIdProps> = (props: ChildrenAndIdProps) => {
    const router = useRouter();

    const { id, children } = props;

    const { orgUserState, templateInteractions } = useLoggedInState();
    const { useSignup } = useConfigState();

    const { ask, dispatchDialogAction, setRefreshCurrentPage } = templateInteractions;
    const currentOrgPermissions = extractPermissionsFromOrgUser(orgUserState);

    const sectionProps: SectionProps<NavGlobalAction> = {
        children,
        sectionKey: SectionKey.globalAction,
        queryResult: useQuery<NavGlobalAction>(NavGlobalActionQuery, {
            variables: { id },
        }),
        sectionHasAnalytics: (data) =>
            analyticsEnabled(data.getActionGroupDistribution.revision.app),
        buildButtonsProps: (data, orgPermissions) => [
            ...buildAppRevisionButtons(
                data.me.orgs,
                data.getActionGroupDistribution.revision.app.tag_manager_account.org,
                data.getActionGroupDistribution.revision.app.tag_manager_account,
                data.getActionGroupDistribution.revision.app.tag_manager_account.org
                    .data_manager_account,
                data.getActionGroupDistribution.revision.app.tag_manager_account.apps,
                data.getActionGroupDistribution.revision.app,
                data.getActionGroupDistribution.revision.app.revisions,
                data.getActionGroupDistribution.revision,
                analyticsEnabled(data.getActionGroupDistribution.revision.app),
                errorTrackingEnabled(data.getActionGroupDistribution.revision.app),
                router,
                orgPermissions,
                useSignup,
                'Global Actions',
            ),
            buildGlobalActionButtonProps(
                router,
                data.getActionGroupDistribution.revision.global_action_group_distributions,
                id,
                data.getActionGroupDistribution.name,
                true,
            ),
        ],
        buildMenuItemsProps: () => [],
        extractOrgUserDetails: (data) =>
            data.getActionGroupDistribution.revision.app.tag_manager_account.org,
        buildBreadcrumbActions: (data) =>
            buildAppRevisionBreadcrumbActions(
                data.getActionGroupDistribution.revision.id,
                data.getActionGroupDistribution.revision.name,
                router,
                currentOrgPermissions,
                dispatchDialogAction,
                setRefreshCurrentPage,
                ask,
                data.getActionGroupDistribution.revision.locked,
            ),
        accountExpireIn: orgUserState?.tagManagerAccount.trialExpiration ?? undefined,
        accountExpired: orgUserState?.tagManagerAccount.trialExpired ?? undefined,
        accountIsTrial: orgUserState?.tagManagerAccount.isTrial ?? undefined,
    };

    return <Section<NavGlobalAction> {...sectionProps} />;
};

export default GlobalActionSection;
