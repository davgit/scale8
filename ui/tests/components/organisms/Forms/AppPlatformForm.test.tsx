import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { FC } from 'react';
import { useMockFormProps } from '../../../componentsMocks/useMockFormProps';
import AppPlatformForm from '../../../../src/components/organisms/Forms/AppPlatformForm';
import { AppPlatformValues } from '../../../../src/dialogPages/tagManager/app/InstallAppPlatform';

const MockForm: FC = () => {
    const formProps = useMockFormProps<AppPlatformValues>({
        platformId: '',
    });

    return <AppPlatformForm {...formProps} availableAppPlatforms={[{ key: 'p1', text: 'p1' }]} />;
};

describe('AppPlatformForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
