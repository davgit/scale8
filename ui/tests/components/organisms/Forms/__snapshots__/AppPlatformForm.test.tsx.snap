// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppPlatformForm form render unchanged 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-s92abg"
  >
    <div
      class="MuiBox-root css-70qvj9"
    >
      <div
        class="MuiBox-root css-1isemmb"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-uix0gv"
      >
        mock form
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiBox-root css-yn9cjk"
    >
      <div
        class="MuiBox-root css-1676va5"
      >
        <form
          class="MuiBox-root css-mjj15b"
        >
          <div
            class="MuiFormControl-root DialogFormField css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
            >
              Platform
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-1ejkwvn-MuiInputBase-root-MuiInput-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-platformId"
                class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                id="mui-component-select-platformId"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="platformId"
                required=""
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-yqkyv6-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiBox-root css-wpygli"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-xcbv7o-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
